pkg/workload/workload.go

```diff
@@ -893,6 +904,34 @@ func HasConditionWithTypeAndReason(w *kueue.Workload, cond *metav1.Condition) bo
 	return false
 }
 
+func HasFailedNodeAnnotation(w *kueue.Workload) bool {
+	if w == nil {
+		return false
+	}
+	annotations := w.GetAnnotations()
+	_, hasFailedNodeAnnotation := annotations[kueuealpha.NodeToReplaceAnnotation]
+	return hasFailedNodeAnnotation
+}
+
+func HasTopologyAssignmentWithFailedNode(w *kueue.Workload) bool {
```

Comment:
```suggestion
func HasTopologyAssignmentWithNodeToReplace(w *kueue.Workload) bool {
```
maybe it is better to do not go into the compentence of the nodefailurecontroller.
We could be replacing for whatever reason.